src/common/datetime/format_time.ts

```diff
@@ -1,29 +1,44 @@
 import { format } from "fecha";
-import { FrontendTranslationData } from "../../data/translation";
+import { FrontendLocaleData, TimeFormat } from "../../data/translation";
 import { toLocaleTimeStringSupportsOptions } from "./check_options_support";
 
+export const useAmPm = (locale: FrontendLocaleData): boolean => {
+  if (locale.time_format === TimeFormat.system) {
+    const test = new Date().toLocaleString(locale.language);
```

Comment:
This is not using the `system`, but the selected language. For system you would need to do:
```
    const test = new Date().toLocaleString();
```